tools/ci/Dockerfile

```diff
@@ -3,19 +3,42 @@ FROM ubuntu:18.04
 ARG UID
 ARG GID
 
-# Install core dependencies
+# Install dependencies
 RUN apt-get update && \
     apt-get install -qq -y --option=Dpkg::Options::=--force-confnew \
         build-essential \
+        clang-format-8 \
+        clang-tidy-8 \
+        cmake \
         gcc \
         git \
+        libglibmm-2.4-dev \
         locales \
+        meson \
         python \
+        python3 \
+        python3-jinja2 \
+        python3-pip \
+        systemd \
         unzip \
         wget \
         xz-utils
 RUN rm -rf /var/lib/apt/lists/*
 
+RUN update-alternatives --install /usr/bin/clang-format clang-format /usr/bin/clang-format-8 1000 && \
+    update-alternatives --install /usr/bin/clang-tidy clang-tidy /usr/bin/clang-tidy-8 1000
+
+# Install googletest package with apt-get above instead when upgrading to Ubuntu >= 19.04.
+RUN tag=release-1.8.1 dir=googletest-$tag && \
```

Comment:
oh no :no_good_man: perhaps we should wait for gtest native to be fixed in Yocto instead?

Reply:
This is for running unit tests on host.

Reply:
Sorry, you said "native".

Looks like googletest is in Yocto master, not Warrior even. ( https://layers.openembedded.org/layerindex/branch/warrior/recipes/?q=googletest , switch to master, 1.8.1 is there.)

Do we want to require the SDK for running unit tests though? This simulates what one would do locally?

Reply:
Could also take down googletest 1.8.1 as a subproject (I added 1.8.1 to the database a while ago): https://wrapdb.mesonbuild.com/gtest

But then it would be downloaded on each build instead of only when the container is built.

Reply:
Do you see any problem with downloading zip from the googletest GitHub page until we upgrade the container to Ubuntu >= 19.04? (I assume we want to hold off with upgrading until the next LTS.)

Reply:
Check this one out: https://github.com/Sarcasm/run-clang-format

Reply:
Git hook might be a good idea as well: https://stackoverflow.com/questions/22866609/can-clang-format-tell-me-if-formatting-changes-are-necessary/22866610#22866610

Reply:
(I think you meant to comment on the other issue.)

> Check this one out: https://github.com/Sarcasm/run-clang-format

A > 300 lines external Python script seems like overkill. Especially when comparing in to the few lines of Python in the current tools/check script related to clang-format.

> Git hook might be a good idea as well

Orthogonal issue. You still need to run it in CI for gating.